src/components/webcams/WebrtcMediaMTX.vue

```diff
@@ -87,140 +96,242 @@ export default class WebrtcRTSPSimpleServer extends Mixins(BaseMixin, WebcamMixi
-        })
-
-        this.ws.onmessage = (msg: MessageEvent) => this.webRtcOnRemoteDescription(msg)
-        this.pc.onicecandidate = (evt: RTCPeerConnectionIceEvent) => this.webRtcOnIceCandidate(evt)
+                for (const candidate of candidatesByMedia[mid]) {
+                    frag += 'a=' + candidate.candidate + '\r\n'
+                }
+            }
 
-        this.pc.oniceconnectionstatechange = () => {
-            if (this.pc === null) return
+            mid++
+        }
 
-            window.console.log('[webcam-rtspsimpleserver] peer connection state:', this.pc.iceConnectionState)
+        return frag
+    }
 
-            this.status = (this.pc?.iceConnectionState ?? '').toString()
+    start() {
+        this.log('requesting ICE servers from ' + this.url)
 
-            if (['failed', 'disconnected'].includes(this.status)) {
+        fetch(this.url, {
+            method: 'OPTIONS',
+        })
+            .then((res) => this.onIceServers(res))
+            .catch((err) => {
+                this.log('error: ' + err)
                 this.scheduleRestart()
-            }
-        }
+            })
+    }
 
-        this.pc.ontrack = (evt: RTCTrackEvent) => {
-            window.console.log('[webcam-rtspsimpleserver] new track ' + evt.track.kind)
-            this.video.srcObject = evt.streams[0]
-        }
+    onIceServers(res: Response) {
+        const iceServers = this.linkToIceServers(res.headers.get('Link'))
+        this.log('ice servers:', iceServers)
+
+        this.pc = new RTCPeerConnection({
+            iceServers,
+        })
 
         const direction = 'sendrecv'
         this.pc.addTransceiver('video', { direction })
         this.pc.addTransceiver('audio', { direction })
 
-        this.pc.createOffer().then((desc: any) => {
-            if (this.pc === null || this.ws === null) return
+        this.pc.onicecandidate = (evt: RTCPeerConnectionIceEvent) => this.onLocalCandidate(evt)
+        this.pc.oniceconnectionstatechange = () => this.onConnectionState()
 
-            this.pc.setLocalDescription(desc)
+        this.pc.ontrack = (evt) => {
+            this.log('new track:', evt.track.kind)
+            this.video.srcObject = evt.streams[0]
+        }
+
+        this.pc.createOffer().then((offer) => this.onLocalOffer(offer))
+    }
 
-            window.console.log('[webcam-rtspsimpleserver] sending offer')
-            this.ws.send(JSON.stringify(desc))
+    // eslint-disable-next-line no-undef
+    onLocalOffer(offer: RTCSessionDescriptionInit) {
+        this.offerData = this.parseOffer(offer.sdp ?? '')
+        this.pc?.setLocalDescription(offer)
+
+        fetch(this.url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/sdp',
+            },
+            body: offer.sdp,
         })
+            .then((res) => {
+                if (res.status !== 201) throw new Error('bad status code')
+                this.eTag = res.headers.get('E-Tag')
```

Comment:
This got changed to `ETag` with MediaMTX 1.1.1 as you can see [here](https://github.com/bluenviron/mediamtx/pull/2401)